Kernel/Devices/Generic/PCSpeakerDevice.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023, Liav A. <liavalb@hotmail.co.il>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/API/ToneGenerationPacket.h>
+#if ARCH(X86_64)
+#    include <Kernel/Arch/x86_64/PCSpeaker.h>
+#endif
+#include <Kernel/Boot/CommandLine.h>
+#include <Kernel/Devices/DeviceManagement.h>
+#include <Kernel/Devices/Generic/PCSpeakerDevice.h>
+#include <Kernel/Sections.h>
+
+namespace Kernel {
+
+UNMAP_AFTER_INIT NonnullRefPtr<PCSpeakerDevice> PCSpeakerDevice::must_create()
+{
+    auto device = MUST(DeviceManagement::try_create_device<PCSpeakerDevice>());
+    return *device;
+}
+
+UNMAP_AFTER_INIT PCSpeakerDevice::PCSpeakerDevice()
+    : CharacterDevice(1, 10)
```

Comment:
Are these just numbered manually? Where do the respective `major` and `minor` numbers come from?

Reply:
Major 1 is for generic devices. I thought about creating some mechanism to expose unix device numbers (Linux does this for dynamic major numbers' allocation). Currently there's none of this.